kv/memberlist/memberlist_client.go

```diff
@@ -1335,7 +1358,7 @@ func (m *KV) MergeRemoteState(data []byte, _ bool) {
 			level.Error(m.logger).Log("msg", "failed to store received value", "key", kvPair.Key, "err", err)
 		} else if newver > 0 {
 			m.notifyWatchers(kvPair.Key)
-			m.broadcastNewValue(kvPair.Key, change, newver, codec)
+			m.broadcastNewValue(kvPair.Key, change, newver, codec, false)
```

Comment:
These messages we're persisting to the local KV that come from external broadcasts are non-CAS by definition? (Only the local mutations are called CAS?)

Reply:
Correct. Currently only CAS operation can modify KV store. There is also Delete operation in [KV client interface](https://github.com/grafana/dskit/blob/47b410a939dfc243124b650db6214e87b0f5a540/kv/client.go#L91-L125), but memberlist implementation [doesn't support it yet](https://github.com/grafana/dskit/blob/4beef469a18ec34fdc12d1fd678b1f0c018de4f4/kv/memberlist/memberlist_client.go#L74-L77). But we should implement it eventually, so perhaps it would be better to call it "local updates", instead of "cas updates". WDYT? I think I'll rename it.

Reply:
Sounds good! CAS is an implementation detail and doesn't really explain that it's a local mod.

Reply:
I've done this change in https://github.com/grafana/dskit/pull/539/commits/979c0311f20e86684374328c29dd386be86e4b30. I've also updated `memberlist_client_messages_in_broadcast_queue` metric to report values for both queues individually.